app/models/cart.rb

```diff
@@ -130,6 +130,22 @@ def add_items_to_checkout(order)
     order
   end
 
+  def shopping_cart_items_equal_order_items?(order)
+    # cart item has quantity, but order item doesn't.
+    # for example: cart and order both have two of the same item which variant_id is 1
+    # cart will look like this: [{variant_id: 1, quantity: 2}]
+    # order will look like this: [{variant_id: 1}, {variant_id: 1}]
+    # before comparing, both need to be converted into the same style
+    cart_items = []
```

Comment:
Lets rename to something like `variant_ids_in_cart`  

cart.cart_items or `self.cart_items = []` is an actual active record setter